src/AppBundle/Services/GuzzleClient.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace AppBundle\Services;
+
+use GuzzleHttp\Client;
+use GuzzleHttp\Exception\TransferException;
+use Symfony\Component\HttpKernel\Exception\HttpException;
+
+class GuzzleClient
```

Comment:
YAGNI, just add FacebookUserProvider and then, when you develop next user provider, you can move common cases in abstraction.